cmd/konvoy-image-wrapper/cmd/wrapper.go

```diff
@@ -368,6 +369,15 @@ func (r *Runner) dockerRun(args []string) error {
 	cmd.Args = append(cmd.Args, args...)
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
+	defer signal.Stop(c)
+	go func() {
+		sig := <-c
+		if signalErr := cmd.Process.Signal(sig); signalErr != nil {
+			fmt.Fprintf(cmd.Stderr, "failed to relay signal %s %v\n", sig.String(), signalErr)
+		}
```

Comment:
I'm not sure that this goroutine finishes running if the user sends no signal. I think it continues to wait to receive a signal from `c` indefinitely.

Reply:
Have you looked at combining `signal.NotifyContext` and `exec.CommandContext`?

Reply:
if the main routine exits this will exit cleanly i don't think this would leak any resources when the whole program exits